core/variable_operations.cpp

```diff
@@ -259,36 +259,74 @@ Variable reverse(Variable var, const Dim dim) {
 /// Return a deep copy of a Variable or of a VariableView.
 Variable copy(const VariableConstView &var) { return Variable(var); }
 
-VariableView nan_to_num(const VariableConstView &var,
-                        const VariableConstView &replacement,
-                        const VariableView &out) {
-  using std::isnan;
+namespace {
+template <typename Op>
+VariableView replace_in_place(const VariableConstView &var,
+                              const VariableConstView &replacement, Op op,
+                              const VariableView &out) {
+
   transform_in_place<std::tuple<double, float>>(
       out, var, replacement,
       scipp::overloaded{
           transform_flags::expect_all_or_none_have_variance,
-          [](auto &a, const auto &b, const auto &repl) {
-            a = isnan(b) ? repl : b;
-          },
+          [&op](auto &a, const auto &b, const auto &repl) { op(a, b, repl); },
           [](units::Unit &a, const units::Unit &b, const units::Unit &repl) {
             expect::equals(b, repl);
             a = b;
           }});
```

Comment:
What is the reason for not putting the whole operator in the element-wise header, so it can be tested in isolation?